components/NavColumn/index.tsx

```diff
@@ -0,0 +1,136 @@
+  LoginButtonsContainer,
+  NameAndStatus,
+  NavColumnContainer,
+  NavColumnHeader,
+  NavLinksContainer,
+  OnboardingButton,
+  Overlay,
+  Profile,
+  ProfileDisplayContainer,
+  ProfileIcon,
+  SignOutButton,
+  SignUpButton,
+} from './styles';
+
+interface NavColumnProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+type NavLink = {
+  name: string;
+  path: string;
+  iconName: IconType;
+};
+
+const navLinks: NavLink[] = [
+  { name: 'View Plants', path: CONFIG.viewPlants, iconName: 'plant' },
+  {
+    name: 'Planting Timeline',
+    path: CONFIG.plantingTimeline,
+    iconName: 'calendar',
+  },
+];
+
+export default function NavColumn({ isOpen, onClose }: NavColumnProps) {
+  const currentPath = usePathname();
+  const { signOut, userId } = useAuth();
+  const router = useRouter();
+  const { profileData, profileReady } = useProfile();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    setIsLoggedIn(userId !== null);
+  }, [userId]);
+
+  const handleSignOut = () => {
+    router.push(CONFIG.login);
+    onClose();
+    signOut();
+  };
+
+  return (
+    <>
+      {isOpen && (
+        <>
+          <Overlay onClick={onClose} $isOpen={isOpen} />
+          <NavColumnContainer>
+            <div>
+              <NavColumnHeader>
+                <div>
+                  {/* empty whitespace for positioning logo and hamburger */}
+                </div>
+                <Link onClick={onClose} href={CONFIG.home}>
+                  <Icon type="logo" />
+                </Link>
+                <HamburgerButton onClick={onClose}>
+                  <HamburgerIcon type="hamburger" />
+                </HamburgerButton>
+              </NavColumnHeader>
+              <NavLinksContainer>
+                {navLinks.map((link: NavLink, key) => (
+                  <NavColumnItem
+                    key={key}
+                    routeName={link.name}
+                    path={link.path}
+                    isSelected={currentPath === link.path}
+                    icon={link.iconName}
+                    onClose={onClose}
+                  />
+                ))}
+              </NavLinksContainer>
+            </div>
+            {!profileReady ? (
+              <div>Loading...</div>
+            ) : isLoggedIn && profileReady && !profileData ? (
+              <OnboardingButton href={CONFIG.onboarding} onClick={onClose}>
+                Go to Onboarding
+              </OnboardingButton>
+            ) : isLoggedIn ? (
+              <ProfileDisplayContainer>
+                <Profile>
+                  <ProfileIcon type="profile" />
+                  <NameAndStatus>
+                    <H3 $color={COLORS.shrub} style={{ fontWeight: 300 }}>
+                      Your Account
+                    </H3>
+                    <H4 $color={COLORS.shrub} style={{ fontWeight: 300 }}>
+                      {formatUserType(profileData?.user_type as UserTypeEnum)}
+                    </H4>
```

Comment:
Your Account should be H4 and User Type should be P3 
also let's not use style prop if possible. instead use the $fontWeight prop directly